src/Sarif.Viewer.VisualStudio/OpenLogFileCommands.cs

```diff
@@ -168,117 +163,80 @@ private async System.Threading.Tasks.Task MenuItemCallbackAsync(object sender, E
+                // the same order it is presented in the class (as typed by the developer).
+                // So keep a small map the filter index to the correct tool format.
+                Dictionary<int, FieldInfo> filterIndexToToolFormatField = new Dictionary<int, FieldInfo>(toolFormatFieldInfos.Length + 1);
 
-                 OpenFileDialog openFileDialog = new OpenFileDialog();
+                foreach (FieldInfo fieldInfo in toolFormatFieldInfos)
+                {
+                    string resourceName = 
+                        fieldInfo.Name.Equals(ToolFormat.None, StringComparison.OrdinalIgnoreCase) ?
+                            nameof(Resources.ImportSARIFFilter) : string.Format(CultureInfo.InvariantCulture, "{0}{1}{2}", FilterResourceNamePrefix, fieldInfo.Name, FilterResourceNameSuffix);
+
+                    string filterString = Resources.ResourceManager.GetString(resourceName, CultureInfo.CurrentCulture);
+                    toolFormatFilters.Add(filterString);
+                    filterIndexToToolFormatField.Add(toolFormatFilters.Count, fieldInfo);
+                }
 
-                openFileDialog.Title = title;
-                openFileDialog.Filter = filter;
-                openFileDialog.RestoreDirectory = true;
+                OpenFileDialog openFileDialog = new OpenFileDialog()
+                {
+                    Title = Resources.ImportLogOpenFileDialogTitle,
+                    Filter = string.Join("|", toolFormatFilters),
+                    RestoreDirectory = true,
+                    Multiselect = false
+                };
 
                 if (!String.IsNullOrWhiteSpace(inputFile))
                 {
                     openFileDialog.FileName = Path.GetFileName(inputFile);
                     openFileDialog.InitialDirectory = Path.GetDirectoryName(inputFile);
                 }
 
+                // Read the user's last tool format selection.
+                int collectionExists;
+                IVsSettingsManager vsSettingsManager = Package.GetGlobalService(typeof(SVsSettingsManager)) as IVsSettingsManager;
+                if (vsSettingsManager != null &&
+                    vsSettingsManager.GetReadOnlySettingsStore((uint)__VsEnclosingScopes.EnclosingScopes_UserSettings, out IVsSettingsStore vsSettingsStore) == VSConstants.S_OK &&
+                    vsSettingsStore.CollectionExists(nameof(SarifViewerPackage), out collectionExists) == VSConstants.S_OK &&
+                    collectionExists != 0 &&
+                    vsSettingsStore.GetString(nameof(SarifViewerPackage), ToolFormatSettingName, out string openLogFileToolFormat) == VSConstants.S_OK)
+                {
+                    int? filterIndex = filterIndexToToolFormatField.
+                        Where(kvp => kvp.Value.Name.Equals(openLogFileToolFormat, StringComparison.Ordinal)).
+                        Select(kvp => kvp.Key).FirstOrDefault();
```

Comment:
Now that I see how it's used, this code suggests that the dictionary should be from string => int, not from int => FieldInfo.